cli/src/action/playlist.rs

```diff
@@ -0,0 +1,244 @@
+// Copyright 2018-2021 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License
+
+use std::fs::File;
+use std::io::Write;
+
+use clap::ArgMatches;
+use transact::families::smallbank::workload::playlist::{
+    generate_smallbank_playlist, process_smallbank_playlist,
+};
+use transact::workload::batch_gen::generate_signed_batches;
+use transact::workload::{submit_batches_from_source, DEFAULT_LOG_TIME_SECS};
+
+use crate::error::CliError;
+
+use super::{create_cylinder_jwt_auth_signer_key, load_cylinder_signer_key, Action};
+
+pub struct CreatePlaylistAction;
+
+impl Action for CreatePlaylistAction {
+    fn run<'a>(&mut self, arg_matches: Option<&ArgMatches<'a>>) -> Result<(), CliError> {
+        let args = arg_matches.ok_or(CliError::RequiresArgs)?;
+
+        let mut output_writer: Box<dyn Write> = match args.value_of("output") {
+            Some(file_name) => File::create(file_name).map(Box::new).map_err(|_| {
+                CliError::ActionError("Unable to create provided output file".to_string())
+            })?,
+            None => Box::new(std::io::stdout()),
+        };
+
+        let workload = args
+            .value_of("workload")
+            .ok_or_else(|| CliError::ActionError("Workload type is required".into()))?;
+
+        match workload {
+            "smallbank" => {
+                let num_accounts = args
+                    .value_of("smallbank_num_account")
+                    .unwrap_or("10")
```

Comment:
This should be defined as a const. Same with all other default values defined in this file.

Reply:
updated